### PR TITLE
RUMM-664 Coil integration with Datadog

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

This PR provide integration support for Coil ImageLoading API 

[Coil][(https://coil-kt.github.io/coil/getting_started/)

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android-coil/src/main/kotlin/com/datadog/android/coil/DatadogCoilRequestListener.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        )
+    }
+
+    // endregion
+
+    // region Internals
+
+    private fun tags(request: ImageRequest): Map<String, Any?> {
+        val requestDataClass = request.data::class.java
+        return when {
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android-coil/src/main/kotlin/com/datadog/android/coil/DatadogCoilRequestListener.kt:43]: Yes

**Review by xgouchet — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -16,6 +16,10 @@ If you are using SDK version `0.2.5` or lower, take a look at the [Migration Gui
 
 If your existing codebase is already using Timber, you can migrate to Datadog easily by using the [dedicated library](dd-sdk-android-timber/README.md).
 
+### Integrating with Coil
+
+If you use Coil to load images in your application, take a look at our [dedicated library](dd-sdk-android-coil/README.md).
```
Comment (ruthnaebeck) [inline README.md:21]:
```suggestion
If you use Coil to load images in your application, take a look at Datadog's [dedicated library](dd-sdk-android-coil/README.md).
```

File: README.md [hunk 3]
```diff
@@ -16,6 +16,10 @@ If you are using SDK version `0.2.5` or lower, take a look at the [Migration Gui
...
+
 ### Integrating with Fresco
 
 If you use Fresco to load images in your application, take a look at our [dedicated library](dd-sdk-android-fresco/README.md).
```
Comment (ruthnaebeck) [inline README.md:25]:
```suggestion
If you use Fresco to load images in your application, take a look at Datadog's [dedicated library](dd-sdk-android-fresco/README.md).
```

File: dd-sdk-android-coil/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Datadog Integration for Coil
+
+## Getting Started 
+
+To include the Datadog integration for [Coil][1] in your project, simply add the
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:5]:
```suggestion
To include the Datadog integration for [Coil][1] in your project, add the
```

File: dd-sdk-android-coil/README.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    maven { url "https://dl.bintray.com/datadog/datadog-maven" }
+}
+
+dependencies {
+    implementation "com.datadoghq:dd-sdk-android:<latest-version>"
+    implementation "com.datadoghq:dd-sdk-android-coil:<latest-version>"
+}
+```
+
+### Initial Setup
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:19]:
```suggestion
### Initial setup
```

File: dd-sdk-android-coil/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+Before you can use the SDK, you need to setup the library with your application
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:21]:
```suggestion
Before using the SDK, set up the library with your application
```

File: dd-sdk-android-coil/README.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+context, your Client token and your Application ID.
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:22]:
```suggestion
context, client token, and application ID. 
```

File: dd-sdk-android-coil/README.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    implementation "com.datadoghq:dd-sdk-android:<latest-version>"
+    implementation "com.datadoghq:dd-sdk-android-coil:<latest-version>"
+}
+```
+
+### Initial Setup
+
+Before you can use the SDK, you need to setup the library with your application
+context, your Client token and your Application ID. 
+To generate a Client token and an Application ID please check **UX Monitoring > RUM Applications > New Application**
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:23]:
```suggestion
To generate a client token and an application ID, check **UX Monitoring > RUM Applications > New Application**
```

File: dd-sdk-android-coil/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        val  logger = Logger.Builder()
+                .setNetworkInfoEnabled(true)
+                .setLogcatLogsEnabled(true)
+                .setDatadogLogsEnabled(true)
+                .build();
+    }
+}
+```
+
+Following Coil's [API documentation][2], you then need to:
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:43]:
```suggestion
Follow Coil's [API documentation][2] to:
```

File: dd-sdk-android-coil/README.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+- Decorate the `ImageRequest.Builder` with the `DatadogCoilRequestListener` whenever you perform an image loading request.
+ 
+ ```kotlin
+     imageView.load(uri){
+        listener(DatadogCoilRequestListener())
+     }
+ ```
+
+Doing so will automatically track Coil's network requests (creating both APM Traces and RUM Resource events), and will also listen for disk cache errors (creating RUM Error events).
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:60]:
```suggestion
This automatically tracks Coil's network requests (creating both APM Traces and RUM Resource events), and listens for disk cache errors (creating RUM Error events).
```

File: dd-sdk-android-coil/README.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+## Contributing
+
+Pull requests are welcome, but please open an issue first to discuss what you
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:64]:
```suggestion
```

File: dd-sdk-android-coil/README.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+would like to change. For more information, read the
```
Comment (ruthnaebeck) [inline dd-sdk-android-coil/README.md:65]:
```suggestion
For details on contributing, read the 
```

File: docs/coil_integration.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+dependencies {
+    implementation "com.datadoghq:dd-sdk-android:x.x.x"
+    implementation "com.datadoghq:dd-sdk-android-coil:x.x.x"
+}
+```
+
+### Coil integration setup
+
+Following Coil's [API documentation][2], you then need to:
```
Comment (ruthnaebeck) [inline docs/coil_integration.md:22]:
```suggestion
Follow Coil's [API documentation][2] to:
```

**Pull request merged**